processor/src/advice/mod.rs

```diff
@@ -7,6 +7,55 @@ use vm_core::{
     AdviceSet, StarkField,
 };
 
+pub trait AdviceProvider {
```

Comment:
Let's add doc comments to this trait - something like the comments we current use for `BaseAdviceProvider`.

Reply:
Done! Reworked a bit some parts of the documentation. Let me know your thoughts